### PR TITLE
Rest controller to review

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/resources/templates/barista/RestController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package #####;
+
+import java.util.List;
+
+import javax.validation.Valid;
+
+/* Import your ${$UpperName} entity here */
```
Comment (rocambille) [inline src/main/resources/templates/barista/RestController.java]: There are to much $ :p

    ${$UpperName}

should be:

    ${UpperName}

File: src/main/resources/templates/barista/RestController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public ${UpperName} read(@PathVariable Long id) {
+        return ${lowerName}Repository.findById(id).get();
+    }
+
+    @PutMapping("/${slugs}/{id}/edit")
+    public ${UpperName} edit(@Valid @ModelAttribute ${UpperName} resource) {
+        return ${lowerName}Repository.save(resource);
+    }
+
+    @PostMapping("/${slugs}/create")
```
Comment (rocambille) [inline src/main/resources/templates/barista/RestController.java]: Should be:

    @PostMapping("/${slugs}")

File: src/main/resources/templates/barista/RestController.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    public List<${UpperName}> browse() {
+        return ${lowerName}Repository.findAll();
+    }
+
+    @GetMapping("/${slugs}/{id}")
+    public ${UpperName} read(@PathVariable Long id) {
+        return ${lowerName}Repository.findById(id).get();
+    }
+
+    @PutMapping("/${slugs}/{id}/edit")
```
Comment (rocambille) [inline src/main/resources/templates/barista/RestController.java]: Should be:

    @PutMapping("/${slugs}/{id}")

File: src/main/resources/templates/barista/RestController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    public ${UpperName} edit(@Valid @ModelAttribute ${UpperName} resource) {
```
Comment (rocambille) [inline src/main/resources/templates/barista/RestController.java]: `@ModelAttribute` can be removed

File: src/main/resources/templates/barista/RestController.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        return ${lowerName}Repository.save(resource);
+    }
+
+    @PostMapping("/${slugs}/create")
+    public ${UpperName} add(@Valid @ModelAttribute ${UpperName} resource) {
```
Comment (rocambille) [inline src/main/resources/templates/barista/RestController.java]: `@ModelAttribute` can be removed

**Review by rocambille — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/barista/RestController.java (hunk 4 above)
Comment (ghost) [reply src/main/resources/templates/barista/RestController.java]: ok  thx

**Pull request merged**